packages/react-router-dom/index.tsx

```diff
@@ -946,9 +954,13 @@ export function useSubmit(): SubmitFunction {
   return useSubmitImpl();
 }
 
-function useSubmitImpl(fetcherKey?: string, routeId?: string): SubmitFunction {
+function useSubmitImpl(
+  fetcherKey?: string,
+  fetcherRouteId?: string
+): SubmitFunction {
   let { router } = useDataRouterContext(DataRouterHook.UseSubmitImpl);
-  let defaultAction = useFormAction();
```

Comment:
This is what causes `fetcher.submit` to be unstable, since it depends on `useLocation`.  Now we only depend on the current `routeId` and we let the router do the relative path resolving